src/obj/read_lexer.rs

```diff
@@ -14,7 +14,7 @@ impl ReadLexer {
     /// Will read from the given `BufRead`as long as it is not EOF.\
     /// When an entity is parsed, the given callback is invoked and the entity is inserted into it as parameter.\
     /// Will return `Ok(())` if successful or an `Error` (if parsing failed).
-    pub fn read_to_end<R: BufRead>(reader: &mut R, callback: impl Fn(Entity)) -> Result<(), Error> {
+    pub fn read_to_end<R: BufRead>(reader: &mut R, mut callback: impl FnMut(Entity)) -> Result<(), Error> {
```

Comment:
You generally want to mutate captured state from this callback, as you will likely be filling some buffers while you are iterating the file.